tests/integration/test_bayesian_optimization.py

```diff
@@ -164,3 +172,67 @@ def build_model(data: Dataset) -> GaussianProcessRegression:
         acquisition_function = acquisition_rule._acquisition_function
         if isinstance(acquisition_function, AcquisitionFunctionClass):
             assert acquisition_function.__call__._get_tracing_count() == 3  # type: ignore
+
+
+@random_seed
+@pytest.mark.parametrize(
+    "num_steps, acquisition_rule",
+    [
+        (5, DiscreteThompsonSampling(1000, 50)),
+    ],
+)
+def test_two_layer_dgp_optimizer_finds_minima_of_michalewicz_function(
+    num_steps: int, acquisition_rule: AcquisitionRule[TensorType, SearchSpace]
+) -> None:
+
+    search_space = Box([0, 0], [pi, pi])
+
+    def build_model(data: Dataset) -> GPfluxModelConfig:
+        epochs = int(2e3)
+        batch_size = 100
+
+        dgp = two_layer_dgp_model(data.query_points)
+
+        def scheduler(epoch: int, lr: float) -> float:
+            if epoch == epochs // 2:
+                return lr * 0.1
+            else:
+                return lr
+
+        optimizer = tf.optimizers.Adam(0.01)
+        fit_args = {
+            "batch_size": batch_size,
+            "epochs": epochs,
+            "verbose": 0,
+            "callbacks": tf.keras.callbacks.LearningRateScheduler(scheduler),
+        }
+
+        config = GPfluxModelConfig(
```

Comment:
perhaps its more direct to simply use `DeepGaussianProcess` wrapper rather than the Config structure?

Reply:
Done